src/bin/server.ts

```diff
@@ -136,12 +136,15 @@ interface Args {
 
   driver.on("driver ready", async () => {
     try {
-      server = new ZwavejsServer(driver, {
-        port: args.port,
-        host: args.host,
-        enableDNSServiceDiscovery: !args["disable-dns-sd"],
-      });
-      await server.start();
+      // only start server if it does not already exist. Avoids port already in use error on driver reset.
+      if (server == null) {
+        server = new ZwavejsServer(driver, {
+          port: args.port,
+          host: args.host,
+          enableDNSServiceDiscovery: !args["disable-dns-sd"],
+        });
+        await server.start();
```

Comment:
Since a restart of the driver requires a full reinitalization of the Z-Wave state by the clients, I would assume that if a server already exists, we should tell it to reset too: purge any state + disconnect all clients.

Reply:
For ease of use we could even consider to just quit the process when we trigger a reset and let it just be automatically restarted ?

Reply:
yes I would agree with that first point. We can quit the process but if the process can purge state and disconnect all clients, it's effectively the same thing right?

Reply:
It is effectively the same thing if it is done 100% right :)

Reply:
does this work in the case where the server is embedded in zwave-js-ui? zwave-js-ui will restart itself on a hard reset, but only when done directly in the UI. If a WS client initiates a hard reset, there's no process to exit in that case because zwave-js-ui is the parent process for the server instance.

Reply:
If Z-Wave JS UI restarts on a hard reset, this call should propagate up and restart when embedded in Z-Wave JS UI, or else things break.

Maybe we should allow passing in a hard reset handler to the constructor. ZJS-UI does it's thing, and our process also just quits.

Reply:
Or we can just emit an `exit` event and add a handler in zwave-js-ui